### ISSUE TITLE
Geany refuses to start because of domain socket

### ISSUE THREAD

**Issue by Rawing** [geany/geany#1411]
The error I'm getting is

> Geany tried to access the Unix Domain socket of another instance running as another user.
> This is a fatal error and Geany will now quit.

I'm aware of [#294](https://github.com/geany/geany/issues/294), but I'm not running geany as root. This error occurs whenever I start an instance of geany, then either start another instance or open a file with geany (through right click -> open with geany).

While an instance of geany is running, `ls -l .config/geany/ | grep socket` shows

`lrwxrwxrwx. 1 root root   60 Feb 26 14:11 geany_socket_Starlight__0 -> /tmp/geany_socket.fcdcdb1b`

If I understand correctly, the owner of this file should be me, not root.

I suspect that this problem is related to the way I've set up my home directory - it's on an NTFS partition. (More accurately, .config/geany is symlinked to a directory on an NTFS partition.)

**Comment by elextr** [comment]
I am not an expert on it, but I understand that NTFS does not store Linux ownerships, so when you mount the NTFS partition you need to specify the user that the mounted NTFS partition will appear to be owned by, or it uses the user who mounted it, in your case I'm guessing root.

**Comment by Rawing** [comment]
I'll try that, but that workaround wouldn't help if I wasn't the only user on this PC. Wouldn't writing the user ID into the file solve the problem?

**Comment by elextr** [comment]
yeah, the problem seems to be that Geany is checking the ownership of the [link](https://github.com/geany/geany/blob/df05fc87120141da8e9fc5b67ac10dd550891e04/src/socket.c#L229), not the socket.  Since the link is there to allow for just your setup, having the configuration directory on a foreign filesystem that may not support sockets, the validity of its permissions and users should not be relied on either.

Should only need to change the line linked above to g_stat().

**Comment by justUmen** [comment]
The same thing happened to me after using geany with root. As user i couldn't start geany again, even after reboot :  

` "Geany tried to access the Unix Domain socket of another instance running as another user. This is a fatal error and Geany will now quit." `

I'm not sure why but I fixed this error with a clean `sudo geany`, closing the wrindow and reopening geany as normal user.

**Comment by elextr** [comment]
@justUmen yes, you almost certainly didn't cleanly close Geany, so the socket owned by root was left there.

**Comment by big-vl** [comment]
sudo -H rm ~/.config/geany/geany_socket_*
